### PR TITLE
buy an album done

### PR DESCRIPTION
**Pull request by jersonlopez**
## Summary

the users can buy albums

## Trello Card

[https://trello.com/b/6AfmahOp/training-expressjs-v2-jerson-l%C3%B3pez-casta%C3%B1o](https://trello.com/b/6AfmahOp/training-expressjs-v2-jerson-l%C3%B3pez-casta%C3%B1o)

### CODE REVIEW

File: app/models/album.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  Album.createModel = album =>
+    Album.create(album).catch(err => {
+      logger.error(err);
+      if (err.name === 'SequelizeUniqueConstraintError') {
+        throw errors.albumOrderError('User already have this album');
+      }
+      throw errors.databaseError(err.message);
+    });
+
+  Album.getAlbumBy = id =>
```
Comment (picollomartin) [inline app/models/album.js]: Remove this unused method

File: app/services/album.js [hunk 2]
```diff
@@ -6,10 +6,12 @@ const axios = require('axios'),
...
   url = config.common.albumsApi.url;
 
-exports.getAll = async source => {
-  const data = await axios.get(`${url}${source}`).catch(err => {
-    logger.error(err);
-    throw errors.albumsApiError(err);
-  });
-  return data.data;
+exports.getAlbums = source => {
+  return axios
```
Comment (picollomartin) [inline app/services/album.js]: Remove brackets and return, also the method implementation is so generic and name isn't

File: test/album.js [hunk 3]
```diff
@@ -76,4 +78,81 @@ describe('albums', () => {
         });
     });
   });
+
+  describe('/albums/:id POST', () => {
+    it('should order an albums without problems because are loged', done => {
```
Comment (picollomartin) [inline test/album.js:88]: Some cases of error are missing

File: app/controllers/album.js [hunk 4]
```diff
@@ -1,13 +1,31 @@
...
       res.status(200).send(albums);
     })
     .catch(next);
 };
+
+exports.buyAnAlbum = async (req, res, next) => {
+  try {
+    const album = await getAlbums(`/albums/${req.params.id}`);
+
+    const newAlbum = {
```
Comment (picollomartin) [inline app/controllers/album.js:20]: What happen if album is missing  ?

File: app/controllers/album.js [hunk 5]
```diff
@@ -1,13 +1,31 @@
...
 
 const errors = require('../errors'),
-  { getAll } = require('../services/album'),
+  { getAlbums } = require('../services/album'),
+  Album = require('../models').albums,
   logger = require('../logger');
 
 exports.albumList = (req, res, next) => {
-  return getAll('/albums')
+  return getAlbums('/albums')
```
Comment (picollomartin) [inline app/controllers/album.js]: Remove brackets and return

**Review by picollomartin — CHANGES_REQUESTED** [review]

File: app/services/album.js (hunk 2 above)
Comment (jersonlopez) [reply app/services/album.js]: but, the method just do get albums, all or by id
what name do you suggest?

File: app/controllers/album.js (hunk 4 above)
Comment (jersonlopez) [reply app/controllers/album.js:20]: the error is catched and the user is informate about de error

File: app/controllers/album.js (hunk 5 above)
Comment (jersonlopez) [reply app/controllers/album.js]: as which

File: app/services/album.js (hunk 2 above)
Comment (picollomartin) [reply app/services/album.js]: With this method i can access to photos too `getAlbums('/photos')`, seems tricky. You could do this method generic method internal to service and expose particular interface

File: app/controllers/album.js (hunk 5 above)
Comment (picollomartin) [reply app/controllers/album.js]: like this `(req, res, next) => getAlbums ....`

File: app/controllers/album.js (hunk 4 above)
Comment (picollomartin) [reply app/controllers/album.js:20]: This is ok, but add some tests cases :ok_hand:

File: app/controllers/album.js (hunk 5 above)
Comment (jersonlopez) [reply app/controllers/album.js]: sorry this comment isn't for here

File: test/album.js (hunk 3 above)
Comment (jersonlopez) [reply test/album.js:88]: as which?

is here. :P

Comment (ghost) [reply test/album.js:88]: Missing album for example, try to buy multiples times same album, etc

File: test/album.js [hunk 6]
```diff
@@ -76,4 +77,152 @@ describe('albums', () => {
...
+        }).then(res => {
+          chai
+            .request(server)
+            .post('/albums/10')
+            .set(config.common.session.header_name, res.headers[config.common.session.header_name])
+            .send({
+              id: res.body.id
+            })
+            .catch(err => {
+              expect(err.response.request.req.res.body.message).to.be.equal(
```
Comment (ghost) [inline test/album.js:186]: Check response status code, should be 404

**Review by ghost — CHANGES_REQUESTED** [review]
Little change

File: test/album.js (hunk 6 above)
Comment (jersonlopez) [reply test/album.js:186]: no, i don't know but always return an 503 code, the difference is the message

**Review by picollomartin — CHANGES_REQUESTED** [review]
Sorry my other account again, check that comment anyway

File: test/album.js (hunk 6 above)
Comment (picollomartin) [reply test/album.js:186]: The message should be that one, check how do you catch error from api

**Pull request merged**